Jantimizer/Histograms/Managers/EquiDepthHistogramManager.cs

```diff
@@ -20,10 +22,18 @@ public EquiDepthHistogramManager(IDataGatherer dataGatherer, int depth) : base(d
 
         protected override async Task AddHistogramForAttribute(string attributeName, string tableName)
         {
-            IDepthHistogram newHistogram = new HistogramEquiDepth(tableName, attributeName, Depth);
-            newHistogram.GenerateHistogramFromSortedGroups(await DataGatherer.GetSortedGroupsFromDb(tableName, attributeName));
-            await Task.Delay(1);
-            AddHistogram(newHistogram);
+            var cacheHisto = await DataGatherer.GetHistogramFromCacheOrNull(tableName, attributeName);
+            if (cacheHisto == null)
+            {
+                IDepthHistogram newHistogram = new HistogramEquiDepth(tableName, attributeName, Depth);
+                newHistogram.GenerateHistogramFromSortedGroups(await DataGatherer.GetSortedGroupsFromDb(tableName, attributeName));
+                string columnHash = await DataGatherer.GetTableAttributeColumnHash(tableName, attributeName);

```

Comment:
A lot of cache-logic is redundant between [EquiDepthVarianceHistogramManager.cs](https://github.com/jamadaha/P6/pull/156/files#diff-b754be31b9ebc605ed42daf0b81c6c4e9af664939a17906cc2c8e0072996bfc4) and [EquiDepthHistogramManager.cs](https://github.com/jamadaha/P6/pull/156/files#diff-5ba7d6027622e400c28a8a0fcd1f7b6bd92c41f9098552014367ed91d456a239)

Maybe some of it can be moved into the baseclass?

Reply:
They both require different implementations, so they cannot be put into the base class.